src/BenchmarkDotNet/Diagnosers/MemoryDiagnoser.cs

```diff
@@ -42,21 +42,7 @@ public IEnumerable<Metric> ProcessResults(DiagnoserResults diagnoserResults)
             if (diagnoserResults.GcStats.Gen2Collections > 0 && Config.DisplayGenColumns)
                 yield return new Metric(GarbageCollectionsMetricDescriptor.Gen2, diagnoserResults.GcStats.Gen2Collections / (double)diagnoserResults.GcStats.TotalOperations * 1000);
 
-            yield return new Metric(AllocatedMemoryMetricDescriptor.Instance, diagnoserResults.GcStats.GetBytesAllocatedPerOperation(diagnoserResults.BenchmarkCase));
-        }
-
-        private class AllocatedMemoryMetricDescriptor : IMetricDescriptor
-        {
-            internal static readonly IMetricDescriptor Instance = new AllocatedMemoryMetricDescriptor();
-
-            public string Id => "Allocated Memory";
-            public string DisplayName => "Allocated";
-            public string Legend => "Allocated memory per single operation (managed only, inclusive, 1KB = 1024B)";
-            public string NumberFormat => "0.##";
-            public UnitType UnitType => UnitType.Size;
-            public string Unit => SizeUnit.B.Name;
-            public bool TheGreaterTheBetter => false;
-            public int PriorityInCategory => GC.MaxGeneration + 1;
+            yield return new Metric(new AllocatedMemoryMetricDescriptor(), diagnoserResults.GcStats.GetBytesAllocatedPerOperation(diagnoserResults.BenchmarkCase));
```

Comment:
this type is state-less and I would prefer to to keep reusing the static instance and avoid allocating the memory. Another reason is that in the past we had few places where we were using references to check for equality and I don't remember if metric descriptors don't do that.

```suggestion
            yield return new Metric(AllocatedMemoryMetricDescriptor.Instance, diagnoserResults.GcStats.GetBytesAllocatedPerOperation(diagnoserResults.BenchmarkCase));
```